asyncapi-cli/src/main/java/io/ballerina/asyncapi/codegenerator/usecase/ExtractIdentifierPathFromSpec.java

```diff
@@ -50,17 +50,33 @@ public String extract() throws BallerinaAsyncApiException {
                             .concat(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER)
                             .concat(" in the Async API Specification")));
         }
-        StringBuilder eventPathString = new StringBuilder(Constants.CLONE_WITH_TYPE_VAR_NAME);
+        StringBuilder eventPathString = new StringBuilder("");
         if (valuesMap.get(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_TYPE)
                 .equals(Constants.X_BALLERINA_EVENT_TYPE_HEADER)) {
-            //TODO: Handle header event path
-        } else { // Defaults to BODY
+            // Handle header event path
+            if (!valuesMap.containsKey(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_NAME)) {
+                throw new BallerinaAsyncApiException(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_NAME
+                        .concat(" attribute is not found within the attribute "
+                                .concat(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER)
+                                .concat(" in the Async API Specification")));
+            }
+            String identifierName = valuesMap.get(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_NAME);
+            if (Constants.BAL_KEYWORDS.stream()
+                    .anyMatch(identifierName::equals)) {
+                eventPathString.append("'").append(identifierName);
+            } else {
+                eventPathString.append(identifierName);
+            }
+        } else if (valuesMap.get(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_TYPE)
+                .equals(Constants.X_BALLERINA_EVENT_TYPE_BODY)) {
+            // Handle body event path
             if (!valuesMap.containsKey(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_PATH)) {
                 throw new BallerinaAsyncApiException(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER_PATH
                         .concat(" attribute is not found within the attribute "
                                 .concat(Constants.X_BALLERINA_EVENT_FIELD_IDENTIFIER)
                                 .concat(" in the Async API Specification")));
             }
+            eventPathString.append(Constants.CLONE_WITH_TYPE_VAR_NAME);
```

Comment:
Is this required?

Reply:
Improved this logic in [b6ae9c1](https://github.com/ballerina-platform/asyncapi-tools/pull/54/commits/b6ae9c1a6a84c715f44126e6cd85ee493177a302)